manager/orchestrator/constraintenforcer/constraint_enforcer.go

```diff
@@ -115,16 +120,36 @@ func (ce *ConstraintEnforcer) rejectNoncompliantTasks(node *api.Node) {
 		// Ensure that the task assigned to the node
 		// still satisfies the resource limits.
 		if t.Spec.Resources != nil && t.Spec.Resources.Reservations != nil {
-			if t.Spec.Resources.Reservations.MemoryBytes > availableMemoryBytes {
+			if t.Spec.Resources.Reservations.MemoryBytes > available.MemoryBytes {
 				removeTasks[t.ID] = t
 				continue
 			}
-			if t.Spec.Resources.Reservations.NanoCPUs > availableNanoCPUs {
+			if t.Spec.Resources.Reservations.NanoCPUs > available.NanoCPUs {
 				removeTasks[t.ID] = t
 				continue
 			}
-			availableMemoryBytes -= t.Spec.Resources.Reservations.MemoryBytes
-			availableNanoCPUs -= t.Spec.Resources.Reservations.NanoCPUs
+			for k, v := range t.Spec.Resources.Reservations.Generic {
+				n, ok := available.Generic[k]
+
+				enough, _ := tpr.HasEnough(n, v)
+				if !ok || !enough {
+					removeTasks[t.ID] = t
+					break
+				}
+			}
+
+			if _, ok := removeTasks[t.ID]; ok {
+				continue
+			}
+
+			available.MemoryBytes -= t.Spec.Resources.Reservations.MemoryBytes
+			available.NanoCPUs -= t.Spec.Resources.Reservations.NanoCPUs
+
+			fakeStore := &api.GenericResource{}
+			for k, v := range t.Spec.Resources.Reservations.Generic {
```

Comment:
I'd think this should look at `t.AssignedGenericResources` rather than `t.Spec.Resources.Reservations.Generic`, so that if an actual resource assigned to a task no longer exists (for example, an external GPU is unplugged), the constraint enforcer can make sure the task gets replaced.